libfwupdplugin/fu-device.c

```diff
@@ -592,36 +605,21 @@ fu_device_remove_private_flag(FuDevice *self, guint64 flag)
  * Since: 1.6.2
  **/
 gboolean
-fu_device_has_private_flag(FuDevice *self, guint64 flag)
+fu_device_has_private_flag(FuDevice *self, const gchar *flag)
 {
-	FuDevicePrivate *priv = GET_PRIVATE(self);
+	FuDevicePrivateFlagItem *item;
+
 	g_return_val_if_fail(FU_IS_DEVICE(self), FALSE);
-#ifndef SUPPORTED_BUILD
-	if (fu_device_private_flag_item_find_by_val(self, flag) == NULL) {
-		g_critical("%s flag 0x%x is unknown -- use fu_device_register_private_flag()",
+	g_return_val_if_fail(flag != NULL, FALSE);
+
+	item = fu_device_private_flag_item_find_by_flag(self, flag);
+	if (item == NULL) {
+		g_critical("%s flag %s is unknown -- use fu_device_register_private_flag()",
```

Comment:
this changes the behavior to show the message even on supported builds.  Is that really what we want?

Reply:
Maybe not; I'll stick the guards back in.